Initium-Odp/src/com/universeprojects/miniup/server/ODPDBAccess.java

```diff
@@ -2949,7 +2978,30 @@ public String cleanCharacterName(String name)
 		name = name.replace("  ", " ");
 		return name;
 	}
-
+	
+	/**
+	 * Returns the CachedEntity object representing the party leader of @partyCode
+	 * Will return null if the partyCode is empty or null (or if for some reason no one in the party is a leader).
+	 * 
+	 * @param ds
+	 * @param partyCode code of the party to grab the leader from
+	 * @return the party leader of the party.
+	 */
+	public CachedEntity getPartyLeader(CachedDatastoreService ds, String partyCode) {
```

Comment:
We should pass in the members of they party as an option, only fetching if the list given is null. Database optimization.